### DISCUSSION TITLE
Wrong find result

### DISCUSSION THREAD

**Discussion by imcm7** [mikro-orm/mikro-orm#4704] [General]
I make a search with

```
        fields: ['id', 'name', 'roles.name'],
        populate: [
          'roles',
        ],
```

And get in result `"permissions": ["1", "2"]`, but I not asked for it.

```
[
  {
    "id": "1",
    "name": "Edit user",
    "roles": [
      {
        "id": "1",
        "name": "Admin",
        "permissions": [
          "1",
          "2"
        ]
      },
      {
        "id": "2",
        "name": "Authenticated",
        "permissions": [
          "1",
          "2"
        ]
      }
    ]
  }
  ```

**Comment by B4nan** [comment]
Please dont ignore the issue template, reproductions are required for bug reports. Also, what driver and version are you on? 

My guess is you got the entities already loaded in the context.

**Reply by imcm7** [reply]
mariadb and @mikro-orm/cli ^5.8.0

And I don't know what that mean `My guess is you got the entities already loaded in the context.`.
I make simply query and indicate what fields I need, but get other fields too.
And I use that https://github.com/imcm7/nuxt-mikroorm-ts

**Reply by B4nan** [reply]
> And I use that [imcm7/nuxt-mikroorm-ts](https://github.com/imcm7/nuxt-mikroorm-ts)

Looks like the project is private, getting 404.

**Reply by imcm7** [reply]
oops, I removed it

**Comment by imcm7** [comment]
@B4nan but for example if I write in you example `nestjs-example-app-master` such query in book controller

```
 @Get(':id')
  async findOne(@Param('id', ParseIntPipe) id: number) {
    return await this.bookRepository.findOneOrFail(id, {
      fields: ['tags.name', 'tags.books.title'],
      populate: ['author', 'tags', 'tags.books'],
    });
  }
  ```
  
  I get such response
  
  ```
  {
  "id":"1",
  "title":"b1", // what title, I don't need it
  "tags":[{
     "id":"1",
      "name":"t1",
     "books":[
            {"id":"1","title":"b1"}
       ]
     },
      {
        "id":"2",
        "name":"t2",
        "books":[{"id":"1","title":"b1"}]
      },
      {
         "id":"3",
         "name":"t3",
         "books":[{"id":"1","title":"b1"}]
     }
  ]}
  ```

**Reply by imcm7** [reply]
And the same is with my first example. Fields condition work incorrectly I think. return data what I not asked for.

**Reply by B4nan** [reply]
Can you try v6? The implicit serialization rules are already changed there (#4138) and it might be actually already addressed.

**Reply by imcm7** [reply]
What to do with that `property 'persist' does not exist on type 'EntityRepository<Author>'.`

**Reply by B4nan** [reply]
https://github.com/mikro-orm/mikro-orm/blob/v6/docs/docs/upgrading-v5-to-v6.md

**Reply by imcm7** [reply]
in v6 looks good
